apps/axistream/src/array.hh

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include "hls_macros.hh"
+
+namespace util {
+
+/*
+ * Simple array type for use in HLS.
+ * This serves as a translatable version of std::array,
+ * for cases when an array with value semantics is needed.
+ * (As opposed to normal C arrays which decay to pointers.)
+ */
```

Comment:
I'm surprised std::array is not directly synthesizable?  I guess it contains malloc?

Reply:
`std::array` doesn't contain `malloc`, there actually is no reason it shouldn't be synthesizable. It's just an array: https://gcc.gnu.org/onlinedocs/gcc-9.2.0/libstdc++/api/a00041_source.html.

I actually never tried synthesizing it before; I just assumed it wouldn't work. I tried it today though and it works fine!

Reply:
nice! seems worth pulling that out then.